test/unit/org/apache/cassandra/auth/StubAuthorizer.java

```diff
@@ -42,34 +42,55 @@ public void clear()
         return perms != null ? perms : Collections.emptySet();
     }
 
-    public void grant(AuthenticatedUser performer,
-                      Set<Permission> permissions,
-                      IResource resource,
-                      RoleResource grantee) throws RequestValidationException, RequestExecutionException
+    public Set<Permission> grant(AuthenticatedUser performer,
+                                 Set<Permission> permissions,
+                                 IResource resource,
+                                 RoleResource grantee) throws RequestValidationException, RequestExecutionException
     {
         Pair<String, IResource> key = Pair.create(grantee.getRoleName(), resource);
-        Set<Permission> perms = userPermissions.get(key);
-        if (null == perms)
+        Set<Permission> oldPermissions = userPermissions.get(key);
+
+        if (oldPermissions == null)
```

Comment:
Not a big deal for a test class, but we could simplify this method a bit by using `Set#computeIfAbsent` and Guava's sets:
```java
{
    Pair<String, IResource> key = Pair.create(grantee.getRoleName(), resource);
    Set<Permission> oldPermissions = userPermissions.computeIfAbsent(key, k -> Collections.emptySet());
    Set<Permission> nonExisting = Sets.difference(permissions, oldPermissions);

    if (!nonExisting.isEmpty())
        userPermissions.put(key, Sets.union(oldPermissions, nonExisting));

    return nonExisting;
}
```

Reply:
Nice I like it.